railgun/internal/proxy/clientgo_cached_proxy_resolver.go

```diff
@@ -176,6 +176,10 @@ func (p *clientgoCachedProxyResolver) DeleteObject(obj client.Object) error {
 // Client Go Cached Proxy Resolver - Private Methods - Cache Server
 // -----------------------------------------------------------------------------
 
+// updateKongAdmin is the function that will be used by the cache server to ultimately make the API
+// call to resolve the current cache to the Kong Admin API configuration endpoint.
+var updateKongAdmin = sendconfig.UpdateKongAdminSimple
+
```

Comment:
-1 to this testing pattern/anti-pattern where the test overwrites some part of the tested logic - due to the [OCP principle](https://en.wikipedia.org/wiki/Open%E2%80%93closed_principle)

What would work better is an Updater interface with a concrete implementation, and a test faking it out.

Reply:
I wasn't quite sure what aspect of SOLID you felt was being offended here, but I settle on working around the lack of function overloading in Go and not making the extensibility provided as publicly available outside of the module as the problem, so I started by taking care of that: a304f29be750e1b900096363d2230fb1d1a5944c

As far as making the `Updater` a interface type as opposed to a function type it doesn't seem like we gain anything for doing that at this point over just keeping the `Updater` as a function type since we're `internal/` and this wont be part of our public API?

Let me know if you feel comfortable with unblocking this given a304f29be750e1b900096363d2230fb1d1a5944c or if you still want to see further change.

Reply:
I wasn't quite sure what aspect of SOLID you felt was being offended here, but I settled on working around the lack of function overloading in Go and not making the extensibility provided as publicly available outside of the module as the problem, so I started by taking care of that: a304f29be750e1b900096363d2230fb1d1a5944c

As far as making the `Updater` a interface type as opposed to a function type it doesn't seem like we gain anything for doing that at this point over just keeping the `Updater` as a function type since we're `internal/` and this wont be part of our public API?

Let me know if you feel comfortable with unblocking this given a304f29be750e1b900096363d2230fb1d1a5944c or if you still want to see further change.

Reply:
> making the Updater a interface type as opposed to a function type

Making this a function type works even better than what I proposed. :+1: